### DISCUSSION TITLE
Fetch result from mutations offline

### DISCUSSION THREAD

**Discussion by Olega95** [zino-app/graphql-flutter#1127] [General]
Hello. I may not have quite a trivial task, but there is no solution to it, due to the fact that you do not have its implementation. Tell me, could you add the following function to the client: when a mutation is made, when the Internet is turned off (I use a self-written Link on offline mutation), in response to a mutation I get data: null. In order to update the cache manually, I need to get the data. Therefore, I have a question, is it possible to add to the client the ability to display mutated data in the update handler in the result when the Internet is turned off? That would be very cool! Thank you!

**Comment by vincenzopalazzo** [comment]
Maybe some caching policy can help you in this case? https://github.com/zino-hofmann/graphql-flutter/blob/fa35fd71fcf9f5b8362bd243372624576d823261/packages/graphql/lib/src/core/policies.dart#L22-L37

Yes off curse, this type of policy don't have any different between connection on/connection off

But is somethings that we can start to discuss a little bit

**Comment by Olega95** [comment]
Yes, unfortunately, Fetch Policy does not affect this issue in any way. I would like that in the absence of an Internet connection, a mutation occurs in the cache and the result is returned to me.

**Reply by vincenzopalazzo** [reply]
I will try to look into it!

**Reply by Olega95** [reply]
In general, the solution is to simply return the submitted data in the desired form, that's all.

**Reply by vincenzopalazzo** [reply]
In general no-cache and only network need to fail in your case because you do not have an internet connection.

>submitted data in the desired form

I don't understand what this means sorry!

In addition, your case is a particular case, so I will try to look more in deep, but without an example this cate time, sorry!
